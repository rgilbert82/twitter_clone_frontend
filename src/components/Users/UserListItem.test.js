import React from 'react';
import { UserListItem } from '.';
import { shallow } from 'enzyme';

describe(UserListItem, () => {
  const userData = { id: 1, name: 'Ryan', username: 'RG', slug: 'rg', token: '12345' };
  let component;

  beforeEach(() => {
    component = shallow(<UserListItem user={userData} />);
  });

  it("renders properly", () => {
    expect(component).toMatchSnapshot();
  });
});
