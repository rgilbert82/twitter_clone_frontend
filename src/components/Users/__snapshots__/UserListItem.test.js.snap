// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`class UserListItem extends _react2.default.Component {
  constructor(props) {
    super(props);
    this.unfollow = this.unfollow.bind(this);
  }

  unfollow() {
    if (!!this.props.unfollowUser) {
      this.props.unfollowUser(this.props.user.id);
    }
  }

  render() {
    const path = '/users/' + this.props.user.slug;
    let follows;
    let unfollow;

    if (!!this.props.user.followers) {
      follows =
      _react2.default.createElement('div', { className: 'user_list_item_follows', __source: { fileName: _jsxFileName, lineNumber: 24 }, __self: this },
        _react2.default.createElement('dl', { __source: { fileName: _jsxFileName, lineNumber: 25 }, __self: this },
          _react2.default.createElement('dt', { __source: { fileName: _jsxFileName, lineNumber: 26 }, __self: this }, 'Followers:'),
          _react2.default.createElement('dd', { __source: { fileName: _jsxFileName, lineNumber: 27 }, __self: this }, this.props.user.followers)),

        _react2.default.createElement('dl', { __source: { fileName: _jsxFileName, lineNumber: 29 }, __self: this },
          _react2.default.createElement('dt', { __source: { fileName: _jsxFileName, lineNumber: 30 }, __self: this }, 'Following:'),
          _react2.default.createElement('dd', { __source: { fileName: _jsxFileName, lineNumber: 31 }, __self: this }, this.props.user.following)));



    }

    if (!!this.props.isCurrentUserPage && !!this.props.isCurrentUserPage()) {
      unfollow = _react2.default.createElement(_reactBootstrap.Button, { className: 'right_float_button', onClick: this.unfollow, __source: { fileName: _jsxFileName, lineNumber: 38 }, __self: this }, 'Unfollow');
    }

    return (
      _react2.default.createElement('div', { className: 'user_list_item', __source: { fileName: _jsxFileName, lineNumber: 42 }, __self: this },
        _react2.default.createElement('div', { className: 'user_list_item_header', __source: { fileName: _jsxFileName, lineNumber: 43 }, __self: this },
          _react2.default.createElement('h3', { __source: { fileName: _jsxFileName, lineNumber: 44 }, __self: this }, _react2.default.createElement(_reactRouterDom.Link, { to: path, __source: { fileName: _jsxFileName, lineNumber: 44 }, __self: this }, this.props.user.name)),
          _react2.default.createElement('aside', { __source: { fileName: _jsxFileName, lineNumber: 45 }, __self: this }, '@', _react2.default.createElement(_reactRouterDom.Link, { to: path, __source: { fileName: _jsxFileName, lineNumber: 45 }, __self: this }, this.props.user.username))),

        follows,
        unfollow));


  }} renders properly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <UserListItem
    user={
        Object {
            "id": 1,
            "name": "Ryan",
            "slug": "rg",
            "token": "12345",
            "username": "RG",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="user_list_item_header"
>
          <h3>
                    <Link
                              replace={false}
                              to="/users/rg"
                    >
                              Ryan
                    </Link>
          </h3>
          <aside>
                    @
                    <Link
                              replace={false}
                              to="/users/rg"
                    >
                              RG
                    </Link>
          </aside>
</div>,
        undefined,
        undefined,
      ],
      "className": "user_list_item",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h3>
              <Link
                            replace={false}
                            to="/users/rg"
              >
                            Ryan
              </Link>
</h3>,
            <aside>
              @
              <Link
                            replace={false}
                            to="/users/rg"
              >
                            RG
              </Link>
</aside>,
          ],
          "className": "user_list_item_header",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <Link
                replace={false}
                to="/users/rg"
>
                Ryan
</Link>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Ryan",
                "replace": false,
                "to": "/users/rg",
              },
              "ref": null,
              "rendered": "Ryan",
              "type": [Function],
            },
            "type": "h3",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "@",
                <Link
                  replace={false}
                  to="/users/rg"
>
                  RG
</Link>,
              ],
            },
            "ref": null,
            "rendered": Array [
              "@",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "RG",
                  "replace": false,
                  "to": "/users/rg",
                },
                "ref": null,
                "rendered": "RG",
                "type": [Function],
              },
            ],
            "type": "aside",
          },
        ],
        "type": "div",
      },
      undefined,
      undefined,
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="user_list_item_header"
>
            <h3>
                        <Link
                                    replace={false}
                                    to="/users/rg"
                        >
                                    Ryan
                        </Link>
            </h3>
            <aside>
                        @
                        <Link
                                    replace={false}
                                    to="/users/rg"
                        >
                                    RG
                        </Link>
            </aside>
</div>,
          undefined,
          undefined,
        ],
        "className": "user_list_item",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h3>
                <Link
                                replace={false}
                                to="/users/rg"
                >
                                Ryan
                </Link>
</h3>,
              <aside>
                @
                <Link
                                replace={false}
                                to="/users/rg"
                >
                                RG
                </Link>
</aside>,
            ],
            "className": "user_list_item_header",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <Link
                  replace={false}
                  to="/users/rg"
>
                  Ryan
</Link>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "Ryan",
                  "replace": false,
                  "to": "/users/rg",
                },
                "ref": null,
                "rendered": "Ryan",
                "type": [Function],
              },
              "type": "h3",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "@",
                  <Link
                    replace={false}
                    to="/users/rg"
>
                    RG
</Link>,
                ],
              },
              "ref": null,
              "rendered": Array [
                "@",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": "RG",
                    "replace": false,
                    "to": "/users/rg",
                  },
                  "ref": null,
                  "rendered": "RG",
                  "type": [Function],
                },
              ],
              "type": "aside",
            },
          ],
          "type": "div",
        },
        undefined,
        undefined,
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
